internal/pkg/aws/cloudwatchlogs/cloudwatchlogs.go

```diff
@@ -185,16 +195,13 @@ func initGetLogEventsInput(opts LogEventsOpts) *cloudwatchlogs.GetLogEventsInput
 // Example: if the prefixes is []string{"a"} and all is []string{"a", "b", "ab"}
 // then it returns []string{"a", "ab"}.
 func filterStringSliceByPrefix(all, prefixes []string) (res []string) {
-	m := make(map[string]bool)
 	for _, candidate := range all {
 		for _, prefix := range prefixes {
 			if strings.HasPrefix(candidate, prefix) {
-				m[candidate] = true
+				res = append(res, candidate)
```

Comment:
This function is updated now so that we can deal with the matching more efficiently.